Source/WebKit/WebProcess/WebPage/FindController.cpp

```diff
@@ -293,11 +289,23 @@ void FindController::findString(const String& string, OptionSet<FindOptions> opt
     }
 
     RefPtr<WebPage> protectedWebPage = m_webPage;
+
+#if ENABLE(IMAGE_ANALYSIS)
+    if (completionHandler) {
+        auto timer = makeUnique<PAL::HysteresisActivity>([protectedWebPage, string, options, maxMatchCount] (PAL::HysteresisState state) {
+            if (state == PAL::HysteresisState::Stopped)
+                protectedWebPage->findController().findString(string, options, maxMatchCount);
+        }, 1_s);
+        m_webPage->corePage()->setImageQueueEmptyHysteresisThenAnalyzeImagesForFindInPage(WTFMove(timer));
```

Comment:
I'll defer to area experts here but I feel my this code might look better if it looked something like:
```
if (m_webPage->corePage()->needsImageAnalysisForFindInPage()) {
    m_webPage->corePage()->analyzeImagesForFindInPage([weakPage = WeakPtr { m_webPage.get() }, string, options, maxMatchCount] {
        if (weakPage)
            weakPage->findController().findString(string, options, maxMatchCount);
    });
}
```

The HysteresisActivity would be fully internal to the ImageAnalysisQueue.
`needsImageAnalysisForFindInPage()` would check something like:
`!analysisOfAllImagesOnPageHasStarted || hysteresisActivityIsStillRunning`

What do you think?

Reply:
Wenson pointed out to me offline that the lambda may get called several times if images get added to the document while we're searching. However, if we're no longer displaying search results, we should definitely no longer call the lambda. I don't think this patch gets this quite right?

In any case, I think it'd be nice if the HysteresisActivity was internal complexity to the AnalysisQueue and here we'd just pass in a lambda that would get called when new images have been processed. And we should probably clear this lambda when we're done searching in page.